dev/com.ibm.ws.org.jboss.weld/src/org/jboss/weld/resources/HotspotReflectionCache.java

```
@@ -0,0 +1,157 @@
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import com.ibm.ws.ffdc.annotation.FFDCIgnore;
+
+/**
+ * {@link ReflectionCache} implementation that works around possible deadlocks in HotSpot:
+ *
+ * @see https://issues.jboss.org/browse/WELD-1169
+ * @see http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=7122142
+ * @see http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=6588239
+ *
+ * @author Jozef Hartinger
+ *
+ */
+public class HotspotReflectionCache extends DefaultReflectionCache {
+
+    private final Class<?> annotationTypeLock;
+
+    public HotspotReflectionCache(TypeStore store) {
+        super(store);
+        try {
+            this.annotationTypeLock = Class.forName("sun.reflect.annotation.AnnotationType");
+        } catch (ClassNotFoundException e) {
+            throw new WeldException(e);
+        }
+    }
+
+    protected static final void dump(Class<?> targetClass) {
+        String className = "HotspotReflectionCache";
+        String methodName = "dump";
+
+        String resourceName = FileLogger.getClassResourceName(targetClass);
+
+        String text = "Class [ " + targetClass.getName() + " ] as [ " + resourceName + " ]";
+        FileLogger.fileLog(className, methodName, text);
+
+        byte[] classBytes;
+        try {
+            classBytes = FileLogger.read(targetClass.getClassLoader(), resourceName);
+        } catch ( IOException e ) {
+            FileLogger.fileStack(className, methodName, "Failed to read [ " + resourceName + " ]", e);
+            return;
+        }
+
+        FileLogger.fileDump(className, methodName, text, classBytes);
+    }
+
+    boolean sentWarning = false;
+
+    @Override
+    @FFDCIgnore(Throwable.class)
+    protected Annotation[] internalGetAnnotations(AnnotatedElement element) {
+        String prop = "";
+
+        try {
+            prop = getSystemProperty("jboss.hotspot.trace.enabled");
+        } catch (Exception ignored) {}
```

FFDCIgnore was for Throwable but you caught Exception. Why do you want to ignore the exception at all anyway?

---

To keep this from affecting anything other than the servers we're adding this unique logging too. 

Doesn't FFDCIgnore also deal with subtypes?